### PR TITLE
feat(cue): time.once one-shot scheduled event + notify action + `cue schedule` CLI

### PR DESCRIPTION
**Pull request by pedramamini**
## Summary

Adds a **`time.once`** Cue event: a one-shot, self-destructing scheduled trigger that fires at a specific wall-clock moment and then removes itself from `cue.yaml`. This is the subsystem for any "do X at a specific time, once" request ("in 20 minutes...", "tomorrow at 9am...", "remind me at 4pm..."). Also adds a **`notify`** action so a scheduled fire can surface a toast through the owning agent instead of running the agent, and a **`maestro-cli cue schedule`** command to author these subscriptions correctly (ISO timestamps, timezone offsets, target_node_key).

## What's included (Phases 01-05)

- **Trigger (Phase 01):** `time.once` trigger source + registry registration. Poll-based at ~30s granularity; required `fire_at` is ISO-8601 with timezone offset (rejected at load time if the offset is missing).
- **Self-destruct + notify (Phase 02):** `cue-self-destruct` YAML rewriter that removes the sub from `cue.yaml`; wired into the completion path (removes on `completed`; also on `failed`/`timeout` unless `self_destruct_on_failure: false`). New main→renderer `notify` bridge and a `notify`-action executor branch.
- **CLI (Phase 03):** `maestro-cli cue schedule` with `--in <offset>` / `--at <timestamp>`, `--notify` / `--sticky` / `--message`, plus `--list` and `--cancel`. Registered as a `cue` subcommand. One-offs default to the \`Tasks\` pipeline.
- **Docs + template var (Phase 04):** documented the `time.once` event across the cue docs and the `_maestro-cue.md` system prompt; wired the `CUE_FIRE_AT` template variable.
- **Validation/normalizer:** `time.once` + `notify` fields validated in the config validator and passed through the config normalizer.
- **Tests (Phase 05):** unit + integration coverage for the trigger source, the `cue schedule` CLI (create/list/cancel), the config normalizer passthrough, and the self-destruct YAML rewriter.

**Missed-fire grace window:** if Maestro is closed when `fire_at` arrives, the sub fires on next startup within `grace_minutes` of `fire_at` (default 360 / 6h); past the window it logs as expired and self-destructs without firing.

## Authoring rule

`time.once` subscriptions should always be created via `maestro-cli cue schedule`, never hand-written - the CLI handles ISO formatting, timezone offsets, and target_node_key generation.

## Stats

31 files changed (+3964 / -68). 5 new source modules, 8 new test files.

## Validation

- `tsc` across all three configs (lint/main/cli): clean
- Prettier: clean
- Full suite via pre-push hook: **30,403 passed**, 108 skipped

## Notes

- Targets `rc` (this work was built on the RC line, rebased onto the current `origin/rc` tip).
- Em/en-dashes converted to hyphens in the cue docs to match the RC style sweep and house style.

## Summary by CodeRabbit

* **New Features**
  * Added one-shot scheduling: create/list/cancel one-time tasks via CLI (relative/absolute times), support for prompt and new notify action (sticky toasts), grace windows, and optional self-destruct on failure; new template variable for fire time.

* **Documentation**
  * Expanded docs, examples, CLI recipes, and schema reference for the new time.once event and ISO-8601 timezone validation.

* **Tests**
  * Added broad test coverage for scheduling, validation, dispatch, notify delivery, self-destruct, and YAML write behaviors.

### CODE REVIEW

**Comment by pedramamini** [comment]
@greptile-apps @coderabbitai please re-review

File: src/main/cue/cue-self-destruct.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		}
+		break;
+	}
+	if (headerLines.length === 0) return '';
+	return headerLines.join('\n') + '\n';
+}
+
+export async function removeSubscriptionFromYaml(
+	projectRoot: string,
+	subscriptionName: string
```
[the remark this answers is not in the document]
Comment (pedramamini) [reply src/main/cue/cue-self-destruct.ts:31]: Intentionally skipping the pipeline-identity refactor here. The practical collision it guards against (two same-named subs in one cue.yaml) is now closed at the source: the CLI 'cue schedule' command rejects any name already present in the file before appending, and `time.once` subs are authored exclusively through that CLI (hand-authoring is explicitly discouraged in the docs/system prompt). Threading `pipeline_name` through `removeSubscriptionFromYaml` would touch the engine completion path, the runtime `requestSelfDestruct` callback, CLI cancel, and ~8 test files for a hand-edited-YAML-only edge case - net negative on the simplicity/risk tradeoff for this PR. Tracking it as a follow-up if same-name-across-pipelines ever becomes a supported authoring pattern for one-shots.
